### ISSUE TITLE
Initialization of KeyManager failed.

### ISSUE THREAD

**Issue by o8ruza8o** [sgan81/apfs-fuse#133]
Thank you so much for this project! I truly appreciate it. 

First, I would like to contribute to your readme as I just got a brand new mbp from work and this is literaly the first thing I am building on it's blank state I could tell you are missing: 

`sudo apt install libfuse3-3 libz-dev`

I won't PR for that small change. 

Also, a bug report as it does not work for me, failing with:
``` bash
Initialization of KeyManager failed.
Unable to load container.
```

And giving no useful info in gdb even with -d 1 option:
```bash
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
Opening device /dev/nvme0n1p2 failed with error Permission denied
Error opening device!
[Inferior 1 (process 20883) exited with code 01]
(gdb) bt
No stack.
```

I would be grateful for your help. Thank you!

**Comment by HLFH** [comment]
Same issue on ArchLinux (that is on BTRFS fs)

> ➜  ~ apfs-fuse /dev/nvme0n1p2 /mnt/apfs
> Initialization of KeyManager failed.
> Unable to load container.
> ➜  ~ uname -r
> 5.10.4-arch2-1
> ➜  ~ fdisk -l
> Disque /dev/nvme0n1 : 1,82 TiB, 2001111162880 octets, 488552530 secteurs
> Modèle de disque : APPLE SSD AP2048M                       
> Unités : secteur de 1 × 4096 = 4096 octets
> Taille de secteur (logique / physique) : 4096 octets / 4096 octets
> taille d'E/S (minimale / optimale) : 4096 octets / 4096 octets
> Type d'étiquette de disque : gpt
> Identifiant de disque : 8D0FADE6-1DB1-458B-ABE3-359F53640C28
> 
> Périphérique       Début       Fin  Secteurs Taille Type
> /dev/nvme0n1p1         6     76805     76800   300M Système EFI
> /dev/nvme0n1p2     76806 244217430 244140625 931,3G APFS Apple
> /dev/nvme0n1p3 244217600 464920575 220702976 841,9G Système de fichiers Linux
> /dev/nvme0n1p4 464920576 488552447  23631872  90,1G Données de base Microsoft

**Comment by ahripanto** [comment]
Are u targeting a Mac with T2 Secure chip?
For Macs with T2, you need to extract the Keybag from the secure enclave.
After that, I have no idea how to combine this to get the hash.
Take a look at this Hashcat post:
https://hashcat.net/forum/thread-9775.html

**Comment by HLFH** [comment]
@ahripanto Yes I do have a Mac Mini 2018 with T2 Secure chip. But it is not encrypted. Are you sure in that case, I need to extract the Keybag from the secure enclave?

**Comment by xxxmonsterxxx** [comment]
> Are u targeting a Mac with T2 Secure chip? For Macs with T2, you need to extract the Keybag from the secure enclave. After that, I have no idea how to combine this to get the hash. Take a look at this Hashcat post: https://hashcat.net/forum/thread-9775.html

Hi, any new solution for mac with t2? Now im trying to recover my data and finding some tools for this.

**Comment by xxxmonsterxxx** [comment]
> @ahripanto Yes I do have a Mac Mini 2018 with T2 Secure chip. But it is not encrypted. Are you sure in that case, I need to extract the Keybag from the secure enclave?

Maybe did you have  already any idea for this thread? :/
